### PR TITLE
A0-3628: Build test node and aleph-e2e-client in parallel

### PR DESCRIPTION
**Pull request by Marcin-Radecki**
# Description

* In https://github.com/Cardinal-Cryptography/aleph-node/blob/main/.github/workflows/_build-test-node-and-runtime.yml  , run build test node and aleph-e2e-client in parallel. This speeds up `PR commit` workflow from 15-16 minutes to 11 minutes. Also, there's a similar quantity speedup in nightly jobs. 
* We do not spawn additional runners since I made 2 small self-hosted jobs one after 
* to keep `secrets` not used in `PR commit` (to support forks PRs), I added an optional `push`  argument to `build-aleph-e2e-client` action

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Testing

Run on demand affected nightly workflows:
* https://github.com/Cardinal-Cryptography/aleph-node/actions/runs/7141784220
* https://github.com/Cardinal-Cryptography/aleph-node/actions/runs/7141739418
* https://github.com/Cardinal-Cryptography/aleph-node/actions/runs/7142022142

### CODE REVIEW

File: .github/workflows/_build-aleph-e2e-client-image.yml [hunk 1]
```diff
@@ -16,11 +21,13 @@ on:
 jobs:
   check-vars-and-secrets:
     name: Check vars and secrets
+    if: ${{ inputs.push == true }}
```
Comment (bitsnops) [inline .github/workflows/_build-aleph-e2e-client-image.yml:24]: Please put `true` in single quotes.  In if conditions, it's string as well.  Been there, had problems.

File: .github/actions/build-and-push-aleph-e2e-client-image/action.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ inputs:
   ecr-public-registry:
     required: true
     description: 'ECR public registry'
+  push:
+    required: false
+    description: 'Set to true to push aleph-e2e-client to ECR'
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/actions/build-and-push-aleph-e2e-client-image/action.yml]: Done

File: .github/actions/build-and-push-aleph-e2e-client-image/action.yml [hunk 3]
```diff
@@ -49,6 +52,7 @@ runs:
           grep aleph_e2e_client) target/release/aleph-e2e-client
 
     - name: Login to Public Amazon ECR
+      if: ${{ inputs.push == true }}
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/actions/build-and-push-aleph-e2e-client-image/action.yml]: Done

File: .github/actions/build-and-push-aleph-e2e-client-image/action.yml [hunk 4]
```diff
@@ -69,7 +73,8 @@ runs:
         docker build --tag ${image_and_tag} -f Dockerfile .
         docker save -o aleph-e2e-client.tar ${image_and_tag}
         echo "image=${image_and_tag}" >> $GITHUB_OUTPUT
-        if [[ '${{ steps.get-ref-properties.outputs.branch }}' -eq 'main' ]]; then
+        if [[ '${{ steps.get-ref-properties.outputs.branch }}' -eq 'main' && \
+              '${{ inputs.push }}' == true ]]; then
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/actions/build-and-push-aleph-e2e-client-image/action.yml]: Done

File: .github/workflows/_build-aleph-e2e-client-image.yml (hunk 1 above)
Comment (Marcin-Radecki) [reply .github/workflows/_build-aleph-e2e-client-image.yml:24]: In this workflow, there's `type: boolean`, should I change it to `type: string` ?

File: .github/workflows/_build-aleph-e2e-client-image.yml [hunk 5]
```diff
@@ -54,3 +61,4 @@ jobs:
           aws-secret-access-key: ${{ secrets.AWS_MAINNET_SECRET_ACCESS_KEY }}
           ecr-public-host: ${{ vars.ECR_PUBLIC_HOST }}
           ecr-public-registry: ${{ vars.ECR_PUBLIC_REGISTRY }}
+          push: ${{ inputs.push }}
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/workflows/_build-aleph-e2e-client-image.yml]: Done

**Review by Marcin-Radecki — COMMENTED** [review]
I've reworked `push` argument to `build-and-push-aleph-e2e-client-image` actions so that it is string. I've left `type: boolean` in workflows though.

File: .github/workflows/_build-aleph-e2e-client-image.yml (hunk 1 above)
Comment (bitsnops) [reply .github/workflows/_build-aleph-e2e-client-image.yml:24]: I know, but you'd still need to compare it to string.  I have checked other workflows but please double check github-actions repo again.

**Review by bitsnops — APPROVED** [review]

**Pull request merged**